src/main/java/org/embulk/input/jira/JiraInputPlugin.java

```diff
@@ -0,0 +1,233 @@
+
+        Schema schema = task.getColumns().toSchema();
+        int taskCount = 1;
+
+        return resume(task.dump(), schema, taskCount, control);
+    }
+
+    @Override
+    public ConfigDiff resume(TaskSource taskSource,
+            Schema schema, int taskCount,
+            InputPlugin.Control control)
+    {
+        control.run(taskSource, schema, taskCount);
+        return Exec.newConfigDiff();
+    }
+
+    @Override
+    public void cleanup(TaskSource taskSource,
+            Schema schema, int taskCount,
+            List<TaskReport> successTaskReports)
+    {
+    }
+
+    @Override
+    public TaskReport run(TaskSource taskSource,
+            Schema schema, int taskIndex,
+            PageOutput output)
+    {
+        PluginTask task = taskSource.loadTask(PluginTask.class);
+        JiraUtil.validateTaskConfig(task);
+        JiraClient jiraClient = getJiraClient();
+        jiraClient.checkUserCredentials(task);
+        try (final PageBuilder pageBuilder = getPageBuilder(schema, output)) {
+            if (isPreview()) {
+                List<Issue> issues = jiraClient.searchIssues(task, 0, PREVIEW_RECORDS_COUNT);
+                for (Issue issue : issues) {
+                    issue.toRecord();
+                    JiraUtil.addRecord(issue, schema, task, pageBuilder);
+                }
+            }
+            else {
+                int currentPage = 0;
+                int totalCount = jiraClient.getTotalCount(task);
+                int totalPage = JiraUtil.calculateTotalPage(totalCount, MAX_RESULTS);
+                LOGGER.info(String.format("Total pages (%d)", totalPage));
+                while (currentPage < totalPage) {
+                    LOGGER.info(String.format("Fetching page %d/%d", (currentPage + 1), totalPage));
+                    List<Issue> issues = jiraClient.searchIssues(task, (currentPage * MAX_RESULTS), MAX_RESULTS);
+                    for (Issue issue : issues) {
+                        issue.toRecord();
+                        JiraUtil.addRecord(issue, schema, task, pageBuilder);
+                    }
+                    currentPage++;
+                }
+            }
+            pageBuilder.finish();
+        }
+        return Exec.newTaskReport();
+    }
+
+    @Override
+    public ConfigDiff guess(ConfigSource config)
+    {
+        // Reset columns in case already have or missing on configuration
+        config.set("columns", new ObjectMapper().createArrayNode());
+        ConfigSource guessConfig = createGuessConfig();
+        GuessExecutor guessExecutor = getGuessExecutor();
+        PluginTask task = config.loadConfig(PluginTask.class);
+        JiraUtil.validateTaskConfig(task);
+        JiraClient jiraClient = getJiraClient();
+        jiraClient.checkUserCredentials(task);
+        List<Issue> issues = jiraClient.searchIssues(task, 0, GUESS_RECORDS_COUNT);
+        issues.forEach(Issue::toRecord);
+        Set<String> uniqueAtrribtes = getUniqueAttributes(issues);
```

Comment:
Nitpick: "Atrribtes", there are several places having the same spelling mistakes.

Reply:
Done